demos/embeddings/watch-your-step-cora-demo.ipynb

```diff
@@ -0,0 +1,449 @@
+    "edgelist[\"label\"] = \"cites\"\n",
+    "\n",
+    "Gnx = nx.from_pandas_edgelist(edgelist, edge_attr=\"label\")\n",
+    "nx.set_node_attributes(Gnx, \"paper\", \"label\")\n",
+    "\n",
+    "feature_names = [\"w_{}\".format(ii) for ii in range(1433)]\n",
+    "column_names =  feature_names + [\"subject\"]\n",
+    "node_data = pd.read_csv(os.path.join(data_dir, \"cora.content\"), sep='\\t', header=None, names=column_names)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating the model\n",
+    "\n",
+    "First, we create a StellarGraph object from our NetworkX graph. From this, we create an `AdjacencyPowerGenerator` which loops through the rows of the first `num_powers` of the adjacency matrix."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/kieranricardo/PycharmProjects/origin-stellargaph/stellargraph/stellargraph/core/experimental.py:88: ExperimentalWarning: AdjacencyPowerGenerator is experimental: lack of unit tests. It may be difficult to use and may have major changes at any time.\n",
+      "  warnings.warn(direct_msg, ExperimentalWarning)\n"
+     ]
+    }
+   ],
+   "source": [
+    "G = StellarGraph(Gnx)\n",
+    "generator = AdjacencyPowerGenerator(G, num_powers=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we use the `WatchYourStep` class to create trainable node embeddings and expected random walks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/kieranricardo/PycharmProjects/origin-stellargaph/stellargraph/stellargraph/core/experimental.py:88: ExperimentalWarning: WatchYourStep is experimental: lack of unit tests. It may be difficult to use and may have major changes at any time.\n",
+      "  warnings.warn(direct_msg, ExperimentalWarning)\n",
+      "/Users/kieranricardo/PycharmProjects/origin-stellargaph/stellargraph/stellargraph/layer/watch_your_step.py:147: UserWarning: embedding_dimension must be even. embedding_dimension will be rounded down to the nearest even number.\n",
+      "  \"embedding_dimension must be even. embedding_dimension will be rounded down to the nearest even number.\"\n",
+      "/Users/kieranricardo/PycharmProjects/origin-stellargaph/stellargraph/stellargraph/core/experimental.py:88: ExperimentalWarning: AttentiveWalk is experimental: lack of unit tests. It may be difficult to use and may have major changes at any time.\n",
+      "  warnings.warn(direct_msg, ExperimentalWarning)\n"
+     ]
+    }
+   ],
+   "source": [
+    "wys = WatchYourStep(\n",
+    "    generator, num_walks=80, \n",
+    "    embedding_dimension=65, \n",
```

Comment:
Are you meaning to trigger the non-even warning here?

Reply:
I was, but perhaps its better to not trigger this in the demo and just mention it in the text..

Reply:
I would personally not demonstrate it here and only have it in the docs for `WatchYourStep` (which currently don't mention it, and so should be updated), but it's up to you: if you do decide to trigger it here, it would be good to mention explicitly in the surrounding text in addition to the docs.